src/screens/IndexScreen.js

```diff
@@ -271,7 +298,18 @@ export const IndexScreen = ({ navigation, route }) => {
                 ListHeaderComponent={
                   <>
                     {!!showFilter && (
-                      <DropdownHeader {...{ query, queryVariables, data, updateListData }} />
+                      <>
+                        <DropdownHeader {...{ query, queryVariables, data, updateListData }} />
+                        {query === QUERY_TYPES.EVENT_RECORDS && (
+                          <View>
+                            <OptionToggle
+                              label={texts.eventRecord.filterByDailyEvents}
+                              onToggle={() => setFilterByDailyEvents((value) => !value)}
```

Comment:
Instead of `(value) => !value`, how about `!filterByDailyEvents`?

Reply:
I implemented it in the same way as for the existing toggle

```
<OptionToggle
  label={texts.pointOfInterest.filterByOpeningTime}
  onToggle={() => setFilterByOpeningTimes((value) => !value)}
  value={filterByOpeningTimes}
/>
```

i also asked myself to do it this way or setting `!filterByDailyEvents` and update the other part with `!filterByOpeningTimes`. but i don't wanted to test these special changes but be sure that everything works the same as before. i also would need to read about react best practices about when to use pre-value passed as param and when to use the state variable directly. sure, in cases where we have the setter method used somewhere without having its state variable, one would use the passed pre-value param. but in cases where we use it in the same component/file? could you make a short research please if you find out more about the best use?

Reply:
Yes, after a little research, I understand better. Thank you.

Reply:
And what is the outcome of the research? Can you tell me please in which situation which usage is preferred?

Reply:
Since it's easy to get confused with all the callbacks and timing, and to avoid any unexpected side-effects, it's best to use the functional updater state setter argument. We can use this method especially when the state before updating the value is unknown.